### PR TITLE
doc: hook up sphinx

### PR DESCRIPTION
**Pull request by whot**
For reference, this is how it looks like: https://whot.github.io/python-dbusmock/
Needs enabling in https://github.com/martinpitt/python-dbusmock/settings/pages, change to `Source: Github Action`.

This is the minimal conversion only, using the readme as index page and with a few rst-ifications in the `DBusMockObject` class (but not in `DBusTestCase` because that would conflict with #184). Once it's in place we can prettify all this one-by-one.

~~A helper `Makefile` is included too to build it locally.~~

### CODE REVIEW

File: Makefile
Comment (martinpitt) [inline Makefile:1]: I really want to avoid a top-level Makefile. This is a Python project, and this may even confuse build system autodetection for distributions/packagers, let alone humans. If there is no pyproject.toml moral equivalent to npm package.json's `"scripts:" {}`, then perhaps a `./build-sphinx.sh`?

Comment (whot) [reply Makefile]: The Makefile is just a helper for us (dbusmock devs), even the CI just calls `sphinx-build -M html doc/ doc/` directly. I've dropped it now.

I'm not sure how much value a locally built sphinx-build is over building it online and referring to readthedocs. So I'm not sure integrating it into `pyproject.toml` is really worth it? If anything we'll just have to spend time debugging it where it doesn't work :)

File: doc/conf.py
Comment (martinpitt) [inline doc/conf.py:12]: This needs to use the dynamic version like [here](https://github.com/martinpitt/python-dbusmock/blob/main/dbusmock/__init__.py#L20).

Comment (martinpitt) [reply doc/conf.py:12]: Hmm, does this even appear anywhere? I don't see it on https://whot.github.io/python-dbusmock/index.html or https://whot.github.io/python-dbusmock/autoapi/dbusmock/index.html or https://whot.github.io/python-dbusmock/search.html?q=0.29.1

Comment (whot) [reply doc/conf.py:12]: I had to grep the source code and yes, it does show up:
`  <title>python-dbusmock &mdash; python-dbusmock 0.29.1 documentation</title>`

Guess with everyone having a thousand tabs open no-one would notice that ;)

Comment (martinpitt) [reply doc/conf.py]: Locally the dynamic import works in the sense of sphinx-build succeeding. I pushed that.

I also added the two new directories to .gitignore.

**Review by martinpitt — APPROVED** [review]
Thank you, this is really nice! I'm waiting for CI to complete. I did the project settings change, so this ought to work after landing.

**Comment by martinpitt** [comment]
Et voilà: https://martinpitt.github.io/python-dbusmock/ :rocket:

**Comment by whot** [comment]
it's like the future! :)

**Pull request merged**
